--- kde-runtime-4.7.3.orig/activitymanager/CMakeLists.txt	2011-05-20 23:32:25.000000000 +0300
+++ kde-runtime-4.7.3/activitymanager/CMakeLists.txt	2011-11-20 21:37:24.000000000 +0200
@@ -55,6 +55,7 @@ include_directories(
 
 set(activity_manager_SRCS
     ActivityManager.cpp
+    ActivityChangedOSD.cpp
     EventProcessor.cpp
     ZeitgeistEventBackend.cpp
     NepomukEventBackend.cpp
@@ -78,6 +79,7 @@ target_link_libraries(
     activity-manager
     ${KDE4_KIO_LIBS}
     ${KDE4_KDEUI_LIBS}
+    ${KDE4_PLASMA_LIBS}
     ${ADDITIONAL_LINK_LIBS}
     )
 
