*** kactivities-4.8.0.orig/service/ActivityManager.cpp	Tue Dec 20 19:49:44 2011
--- kactivities-4.8.0/service/ActivityManager.cpp	Sun Mar  4 11:16:38 2012
***************
*** 77,82 ****
--- 77,83 ----
              QHash < KUrl, SharedInfo::ResourceData > & _resources
          )
      : config("activitymanagerrc"),
+       activity_changed_osd(0),
        windows(_windows),
        resources(_resources),
  #ifdef HAVE_NEPOMUK
***************
*** 93,98 ****
--- 94,101 ----
      Nepomuk::ResourceManager::instance()->init();
  #endif
  
+     activity_changed_osd = new ActivityChangedOSD(parent);
+     
      // Initializing config
      connect(&configSyncTimer, SIGNAL(timeout()),
               this, SLOT(configSync()));
***************
*** 275,280 ****
--- 278,285 ----
  
          q->StartActivity(id);
  
+         activity_changed_osd->activityChanged(id);
+ 
          currentActivity = id;
          mainConfig().writeEntry("currentActivity", id);
  
