*** kactivities-4.8.0.orig/service/ActivityManager_p.h	Fri Dec  2 23:16:56 2011
--- kactivities-4.8.0/service/ActivityManager_p.h	Fri Mar  9 19:57:01 2012
***************
*** 76,81 ****
--- 76,83 ----
      QHash < WId, SharedInfo::WindowData > & windows;
      QHash < KUrl, SharedInfo::ResourceData > & resources;
      
+     QHash < QString, int > activitiesDesktops;
+ 
  public:
      void initConifg();
  
