--- kde-runtime-4.7.0.orig/activitymanager/ActivityManager_p.h	2011-05-20 23:32:25.000000000 +0300
+++ kde-runtime-4.7.0/activitymanager/ActivityManager_p.h	2011-08-15 16:04:18.000000000 +0300
@@ -77,6 +77,7 @@ public:
 public:
     void initConifg();
 
+    KConfigGroup activitiesDesktopsConfig();
     KConfigGroup activitiesConfig();
     KConfigGroup mainConfig();
     QString activityName(const QString & id);

