*** kactivities-4.8.0/service/ActivityManager.cpp	Thu Feb 23 21:38:08 2012
--- kactivities-4.8.0.orig/service/ActivityManager.cpp	Tue Dec 20 19:49:44 2011
***************
*** 272,279 ****
          //             Event::Opened, Event::User
          //         );
          // }
-         
-         activitiesDesktops[currentActivity] = KWindowSystem::currentDesktop();
  
          q->StartActivity(id);
  
--- 272,277 ----
***************
*** 286,299 ****
      // kDebug() << (void*) SharedInfo::self() << "Rankings << shared info";
      SharedInfo::self()->setCurrentActivity(id);
      emit q->CurrentActivityChanged(id);
-     
-     if (activitiesDesktops.contains(id)) {
-         int desktopId = activitiesDesktops[id];
-         
-         if (desktopId <= KWindowSystem::numberOfDesktops())
-             KWindowSystem::setCurrentDesktop(desktopId);
-     }
-     
      return true;
  }
  
--- 284,289 ----
