*** kactivities-4.8.1.orig/service/ActivityManager_p.h	Fri Dec  2 23:16:56 2011
--- kactivities-4.8.1/service/ActivityManager_p.h	Mon Mar 12 19:37:04 2012
***************
*** 76,81 ****
--- 76,83 ----
      QHash < WId, SharedInfo::WindowData > & windows;
      QHash < KUrl, SharedInfo::ResourceData > & resources;
      
+     QHash < QString, int > activitiesDesktops;
+ 
  public:
      void initConifg();
  
