*** kactivities-4.8.0.orig/service/ActivityManager.cpp	Tue Dec 20 19:49:44 2011
--- kactivities-4.8.0/service/ActivityManager.cpp	Fri Mar  9 19:58:05 2012
***************
*** 273,278 ****
--- 273,280 ----
          //         );
          // }
  
+         activitiesDesktops[currentActivity] = KWindowSystem::currentDesktop();
+ 
          q->StartActivity(id);
  
          currentActivity = id;
***************
*** 284,289 ****
--- 286,299 ----
      // kDebug() << (void*) SharedInfo::self() << "Rankings << shared info";
      SharedInfo::self()->setCurrentActivity(id);
      emit q->CurrentActivityChanged(id);
+     
+     if (activitiesDesktops.contains(id)) {
+         int desktopId = activitiesDesktops[id];
+         
+         if (desktopId <= KWindowSystem::numberOfDesktops())
+             KWindowSystem::setCurrentDesktop(desktopId);
+     }
+     
      return true;
  }
  
