*** kactivities-4.8.1.orig/service/ActivityManager.cpp	Tue Dec 20 19:49:44 2011
--- kactivities-4.8.1/service/ActivityManager.cpp	Wed Mar 14 21:24:22 2012
***************
*** 113,118 ****
--- 113,122 ----
          }
      }
  
+     foreach (const QString & activity, activitiesDesktopsConfig().keyList()) {
+         activitiesDesktops[activity] = activitiesDesktopsConfig().readEntry(activity).toInt();
+     }
+ 
      syncActivitiesWithNepomuk();
  
      currentActivity = mainConfig().readEntry("currentActivity", QString());
***************
*** 226,231 ****
--- 230,240 ----
      return KConfigGroup(&config, "main");
  }
  
+ KConfigGroup ActivityManagerPrivate::activitiesDesktopsConfig()
+ {
+     return KConfigGroup(&config, "activitiesDesktops");
+ }
+ 
  void ActivityManagerPrivate::ensureCurrentActivityIsRunning()
  {
      QStringList runningActivities = q->ListActivities(ActivityManager::Running);
***************
*** 273,278 ****
--- 282,290 ----
          //         );
          // }
  
+         activitiesDesktops[currentActivity] = KWindowSystem::currentDesktop();
+         activitiesDesktopsConfig().writeEntry(currentActivity, QString::number(KWindowSystem::currentDesktop()));
+ 
          q->StartActivity(id);
  
          currentActivity = id;
***************
*** 284,289 ****
--- 296,309 ----
      // kDebug() << (void*) SharedInfo::self() << "Rankings << shared info";
      SharedInfo::self()->setCurrentActivity(id);
      emit q->CurrentActivityChanged(id);
+ 
+     if (activitiesDesktops.contains(id)) {
+         int desktopId = activitiesDesktops[id];
+ 
+         if (desktopId <= KWindowSystem::numberOfDesktops())
+             KWindowSystem::setCurrentDesktop(desktopId);
+     }
+ 
      return true;
  }
  
