*** kactivities-4.8.1.orig/service/ActivityManager.cpp	Tue Dec 20 19:49:44 2011
--- kactivities-4.8.1/service/ActivityManager.cpp	Sun Mar 18 11:13:59 2012
***************
*** 226,231 ****
--- 226,236 ----
      return KConfigGroup(&config, "main");
  }
  
+ KConfigGroup ActivityManagerPrivate::activitiesDesktopsConfig()
+ {
+     return KConfigGroup(&config, "activitiesDesktops");
+ }
+ 
  void ActivityManagerPrivate::ensureCurrentActivityIsRunning()
  {
      QStringList runningActivities = q->ListActivities(ActivityManager::Running);
***************
*** 273,278 ****
--- 278,285 ----
          //         );
          // }
  
+         activitiesDesktopsConfig().writeEntry(currentActivity, QString::number(KWindowSystem::currentDesktop()));
+ 
          q->StartActivity(id);
  
          currentActivity = id;
***************
*** 284,289 ****
--- 291,304 ----
      // kDebug() << (void*) SharedInfo::self() << "Rankings << shared info";
      SharedInfo::self()->setCurrentActivity(id);
      emit q->CurrentActivityChanged(id);
+ 
+     if (activitiesDesktopsConfig().hasKey(id)) {
+         int desktopId = activitiesDesktopsConfig().readEntry(id).toInt();
+ 
+         if (desktopId <= KWindowSystem::numberOfDesktops() && desktopId >= 0)
+             KWindowSystem::setCurrentDesktop(desktopId);
+     }
+ 
      return true;
  }
  
