--- kde-workspace-4.7.3/kwin/workspace.h	2011-09-01 23:47:10.000000000 +0300
+++ kde-workspace-4.7.3.modified/kwin/workspace.h	2011-11-19 09:32:00.000000000 +0200
@@ -272,12 +272,20 @@ public:
      * @returns The ID of the current desktop.
      */
     int currentDesktop() const;
+    
+    /**
+     * @returns the ID of the previous desktop.
+     */
+    int previousDesktop() const;
+    
     /**
      * Set the current desktop to @a current.
      * @returns True on success, false otherwise.
      */
     bool setCurrentDesktop(int current);
 
+    bool setPreviousDesktop(int current);
+    
     /**
      * Generate a desktop layout from EWMH _NET_DESKTOP_LAYOUT property parameters.
      */
@@ -323,6 +331,8 @@ private:
     QSize desktopGridSize_;
     int* desktopGrid_;
     int currentDesktop_;
+    int previousDesktop_;
+    bool flagToggleDesktops;
     QString activity_;
     QStringList allActivities_;
 
@@ -571,6 +581,8 @@ public slots:
     void slotSwitchDesktopLeft();
     void slotSwitchDesktopUp();
     void slotSwitchDesktopDown();
+    
+    void slotToggleBetweenDesktops();
 
     void slotSwitchToDesktop();
     //void slotSwitchToWindow( int );
@@ -1137,6 +1149,11 @@ inline int Workspace::currentDesktop() c
     return currentDesktop_;
 }
 
+inline int Workspace::previousDesktop() const
+{
+    return previousDesktop_;
+}
+
 inline int Workspace::desktopAtCoords(QPoint coords) const
 {
     return desktopGrid_[coords.y() * desktopGridSize_.width() + coords.x()];
